packages/react-query/src/shared/polymorphism/queryLike.ts

```diff
@@ -20,6 +23,14 @@ export type QueryLike<
     variables: inferProcedureInput<TProcedure>,
     opts?: InferQueryOptions<TConfig, TProcedure, any>,
   ) => InferQueryResult<TConfig, TProcedure>;
+
+  useSuspenseQuery: (
+    variables: inferProcedureInput<TProcedure>,
+    opts?: InferQueryOptions<TConfig, TProcedure, any>,
+  ) => UseTRPCSuspenseQueryResult<
+    inferProcedureOutput<TProcedure>,
+    TRPCClientErrorLike<TConfig>
+  >;
```

Comment:
Since suspense:true is gone on useQuery now, my own codebase needs this to continue using suspense in polymorphic components.

Longer term I do think we should fine a way to drop complexity of the polymorphism types and give them a rebrand at bear minimum, since they've been historically a bit fragile